### PR TITLE
Update Namelist documentation for all namelist options

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
Several UFS Apps use undocumented namelist settings in the UPP Read the Docs documentation. Specifically, I made these changes:

* Added explanations for all namelist options used in the UPP code, and based the descriptions on the comments in the code where they were available.
* The additions follow the documentation style of the ufs-weather-model namelist documentation. That means each entry is displayed in a table format, and information about type and default value is included.
*  I modified the style of the rst file to use long lines in paragraphs instead of wrapping, consistent with @gspetro-NOAA's documentation style preferences for supportability. Specifically, when diffs are proposed in PRs, GitHub's highlighting will draw the eye to exact differences made without concern for re-wrapping lines.
* I updated the EMC URL because it failed during the docs build.

Fixes Issue #949

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
@christinaholtNOAA Can you open a new UPP issue at [here](https://github.com/NOAA-EMC/UPP/issues) to link this PR?

File: doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst [hunk 1]
```diff
@@ -44,41 +42,157 @@ UFS Unified Model Variables
...
+   * - ``grib``
+     - Grib type (Note that UPP only supports Grib2 currently)
+     - character(5)
+     - grib2
+   * - ``ioform``
+     - Input file format. Choices: binary, binarympiio, binarynemsio, netcdf, netcdfpara
+     - character(len=20)
+     - n/a
+   * - ``modelname``
+     - Model name used by UPP internally (e.g., FV3R for LAM, GFS for GFS, NCAR for WRF)
```
Comment (WenMeng-NOAA) [inline doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst]: @christinaholtNOAA I would suggest update as:
e.g., FV3R for RRFS, 3DRTMA, HAFS; GFS for GFS and GEFS; RAPR for RAP and HRRR; NMM for NAM
** 'NCAR' is not a valid UPP internal model name any more.

File: doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst [hunk 2]
```diff
@@ -44,41 +42,157 @@ UFS Unified Model Variables
...
+   * - ``filenameflux``
+     - Name of input physics file; name of 2-D model output file with physics and surface fields.
+     - character(len=256)
+     - n/a
+   * - ``grib``
+     - Grib type (Note that UPP only supports Grib2 currently)
+     - character(5)
+     - grib2
+   * - ``ioform``
+     - Input file format. Choices: binary, binarympiio, binarynemsio, netcdf, netcdfpara
```
Comment (WenMeng-NOAA) [inline doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst]: @christinaholtNOAA The available model input formats are binarynemsio, netcdf,

**Comment by WenMeng-NOAA** [comment]
@christinaholtNOAA Can you sync your branch with the latest UPP develop?

**Comment by christinaholtNOAA** [comment]
Sorry I am falling behind on your requests. I will open the ticket, apply the suggested changes, and update the branch by tomorrow. Thanks for guiding me through the process.

File: doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst [hunk 3]
```diff
@@ -44,41 +42,157 @@ UFS Unified Model Variables
...
-
-#. ``DateStr``: Forecast valid time (not model start time) in YYYY-MM-DD_HH:00:00 format (the forecast time
-   desired to be post-processed).
-
-#. ``MODELNAME``: Model used (GFS, FV3R --- regional FV3, also known as the :term:`LAM`).
-
-#. ``fileNameFlux``: Name of the FV3 (surface) output file to be post-processed.
-
-#. ``fileNameFlat``: Name of configuration file (``postxconfig-NT.txt``)
+The ``&model_inputs`` section parameterizes choices about the set of model output files that will be used for the UPP
```
Comment (gspetro-NOAA) [inline doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst]:
```suggestion
The ``&model_inputs`` section parameterizes choices about the set of model output files that will be used for the UPP. 
```

File: doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst [hunk 4]
```diff
@@ -44,41 +42,157 @@ UFS Unified Model Variables
...
+     - n/a
+   * - ``fileName``
+     - Name of input dynamics file; name of full 3-D model output file.
+     - character(len=256)
+     - n/a
+   * - ``filenameflat``
+     - Input configuration text file defining the requested fields.
+     - character(len=256)
+     - postxconfig-NT.txt
+   * - ``filenameflux``
```
Comment (gspetro-NOAA) [inline doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst:71]:
```suggestion
   * - ``fileNameFlux``
```

File: doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst [hunk 5]
```diff
@@ -44,41 +42,157 @@ UFS Unified Model Variables
...
+
+.. list-table:: *Description of the &model_inputs namelist section.*
+   :widths: 15 40 20 20
+   :header-rows: 1
+
+   * - Variable Name
+     - Description
+     - Data Type
+     - Default Value
+   * - ``datestring``
```
Comment (gspetro-NOAA) [inline doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst]:
```suggestion
   * - ``dateStr``
```

File: doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst [hunk 6]
```diff
@@ -44,41 +42,157 @@ UFS Unified Model Variables
...
+
+.. list-table:: *Description of the &nampgb namelist section.*
+   :widths: 25 40 20 15
+   :header-rows: 1
+
+   * - Variable Name
+     - Description
+     - Data Type
+     - Default Value
+   * - ``aqf_on bool``
```
Comment (gspetro-NOAA) [inline doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst]:
```suggestion
   * - ``aqf_on``
```

File: doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst (hunk 2 above)
Comment (christinaholtNOAA) [reply doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst]: Thanks for narrowing that down. This list I am including here came from scanning the code for all instances of `ioform` checked against a string. Is there a plan to remove that logic?

**Comment by christinaholtNOAA** [comment]
@gspetro-NOAA, I am curious if you'd be okay with a standardized lowercase-only set of variables. Given that the capitalization strategy of these variables doesn't seem to have a rhyme or reason and that Fortran doesn't care, it seems like it might be more readable for users trying to get their namelists "right."

**Comment by gspetro-NOAA** [comment]
@christinaholtNOAA I think my main concern is just that the variables appear in the UPP code later in camel casing. I'm not sure what impact, if any, lowercasing things would have down the line. Lowercasing seems beneficial for uniformity and readability, but less expert users might be confused if later on they see camel-cased variables. So I can see plusses and minuses for both. @WenMeng-NOAA @FernandoAndrade-NOAA do you have any opinion?

File: doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst (hunk 2 above)
Comment (WenMeng-NOAA) [reply doc/UserGuide/BuildingRunningTesting/InputsOutputs.rst]: @christinaholtNOAA Yes, the model formats binary, binarympiio and binarynesio and netcdpara are not supported in UFS community.  Cleaning up these legacy codes in UPP code is in our to-do-list. At this time, we don't want to mislead community users with the UPP documentation.

**Comment by FernandoAndrade-NOAA** [comment]
> @christinaholtNOAA I think my main concern is just that the variables appear in the UPP code later in camel casing. I'm not sure what impact, if any, lowercasing things would have down the line. Lowercasing seems beneficial for uniformity and readability, but less expert users might be confused if later on they see camel-cased variables. So I can see plusses and minuses for both. @WenMeng-NOAA @FernandoAndrade-NOAA do you have any opinion?

In the long run, I'd say uniformity and readability may be the bigger benefit than avoiding temporary confusion with users, even if I am partial to camel casing.

**Comment by christinaholtNOAA** [comment]
@WenMeng-NOAA Do you have any thoughts here? It seems like we might be leaning for uniformity to go with lower case in the docs, if you're in agreement.

**Comment by WenMeng-NOAA** [comment]
> @WenMeng-NOAA Do you have any thoughts here? It seems like we might be leaning for uniformity to go with lower case in the docs, if you're in agreement.

@christinaholtNOAA Fortran coding is case insensitive. From documentation uniformity and readability perspective, using lower case would be good.

**Comment by WenMeng-NOAA** [comment]
@gspetro-NOAA We will wrap up this PR for testing. Please let us know if you have more comments.

**Comment by WenMeng-NOAA** [comment]
@FernandoAndrade-NOAA Conducting the UPP RTs on either Hera or Orion would be sufficient.

**Comment by FernandoAndrade-NOAA** [comment]
> @FernandoAndrade-NOAA Conducting the UPP RTs on either Hera or Orion would be sufficient.

Hera RTs finished, this PR should be ready for merging.

**Review by FernandoAndrade-NOAA — APPROVED** [review]

**Review by WenMeng-NOAA — APPROVED** [review]

**Pull request merged**